frontend/apps/filemanager/filemanagerutil.lua

```diff
@@ -29,12 +29,36 @@ function filemanagerutil.abbreviate(path)
     return path
 end
 
--- Purge doc settings in sidecar directory,
+-- Purge doc settings in sidecar directory
 function filemanagerutil.purgeSettings(file)
     local file_abs_path = util.realpath(file)
     if file_abs_path then
         DocSettings:open(file_abs_path):purge()
     end
 end
 
+-- Purge doc settings except kept
+function filemanagerutil.purgeViewSettings(file)
+    local settings_kept = {
+        "bookmarks",
+        "bookmarks_sorted",
+        "bookmarks_version",
+        "cre_dom_version",
+        "highlight",
+        "highlights_imported",
+        "last_page",
+        "last_xpointer",
+    }
+    local file_abs_path = util.realpath(file)
+    if file_abs_path then
+        local doc_settings = DocSettings:open(file_abs_path)
+        for k in pairs(doc_settings.data) do
+            if not require("util").arrayContains(settings_kept, k) then
```

Comment:
This is a bit ugly/expensive (doesn't matter much in this context, so, just ugly :)
I would build a hash table from the array, and just use the hash table:
local settings_to_keep = {}
`for _, k in ipairs(settings_kept ) do settings_to_keep[k] = true end`
and here just:
`if not settings_to_keep[k] then`
or just define this little bit uglier:
```
local settings_kept = {
        "bookmarks" = true
        "bookmarks_sorted" = true
        "bookmarks_version" = true
```

Reply:
Thanks